### PR TITLE
Object manipulation 3

### PR DESCRIPTION
**Pull request by annaaksenov**
object-manipulation-3 - Anna Ak.

### CODE REVIEW

File: object-manipulation-3/main.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+//   function game(players, hand) {
+//  const shuff = _.shuffle(deck);
+//  const deal = _.chunk(shuff, [2]);
+
+//  players.push(deal)[1];
+//  console.log(players);
+//  shuffle the deck. w/ _.shuffle();
+//  deal 2 cards to each player. how idk. w/ _.chunk().
+// push? the deck into 'hand' arrays... for...in loop?
```
Comment (kurtyazdizadeh) [inline object-manipulation-3/main.js:269]: to deal cards to the player, you can take the shuffled array of cards and either use other lodash array methods like `_first` or a native javascript array method like `splice`

**Review by kurtyazdizadeh — CHANGES_REQUESTED** [review]

File: object-manipulation-3/main.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    players[i].hand = [...shuffChunk.splice(0, 1)];
+    if (players[i].hand === ranks[0]) {
+      score += 11;
+    } else if (players[i].hand === ranks[10] || ranks[11] || ranks[12]) {
+      score += 10;
+    }
+  }
+
+  return score;
+}
```
Comment (TimelikeClosure) [inline object-manipulation-3/main.js:35]: It seems like your `shuffleDeck` function is attempting to do more that just shuffling the deck. I suggest only trying to deal cards from the `deal` function.

File: object-manipulation-3/main.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+shuffleDeck(createDeck());
+console.log(shuffleDeck(createDeck()));
+
+/*
+ function deal(eachOne) {
+  for (let playerCount = 0; playerCount < 4; playerCount++) {
+    players[playerCount].hand = shuffleDeck.splice(0, 3);
+  }
+  return players;
+}
```
Comment (TimelikeClosure) [inline object-manipulation-3/main.js:45]: I suggest having your `deal` function take two parameters: the deck, and the players.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by kurtyazdizadeh — APPROVED** [review]
Looks like it works! Nice job, approved 👍

**Pull request merged**